solr/core/src/test/org/apache/solr/update/DirectUpdateHandlerTest.java

```diff
@@ -136,22 +118,39 @@ public void testBasics() {
     assertQ(req("q", "id:5"), "//*[@numFound='0']");
     assertQ(req("q", "id:6"), "//*[@numFound='0']");
 
-    long newAdds = ((Gauge<Number>) metrics.get(addsName)).getValue().longValue();
-    long newCumulativeAdds = ((Meter) metrics.get(cumulativeAddsName)).getCount();
-    assertEquals("new adds", 2, newAdds - adds);
-    assertEquals("new cumulative adds", 2, newCumulativeAdds - cumulativeAdds);
+    assertEquals(
+        "Did not have the expected number of pending adds",
```

Comment:
as you went through the effort of writng these messages -- fine.  But I wouldn't waste your time; it's optional.